lab-davide/test/note-router-test.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+
+process.env.MONGO_URI = 'mongodb://localhost/test';
+
+//npm module//
+const debug = require('debug')('note:note-tests');
+const expect = require('chai').expect;
+const request = require('superagent-use');
+const superPromise = require('superagent-promise-plugin');
+
+
+request.use(superPromise);
+
+
+//app modules//
+const server = require('../server');
+const noteCrud = require('../lib/note-crud');
+
+//required modules dependent on global env//
+const port = process.env.PORT || 3000;
+const baseUrl = `http://localhost:${port}/api/note`;
+
+
+//server is running//
+describe('testing module note-router', function() {
+  debug('hitting module note router');
+  before((done) => {
+    if (!server.isRunning) {
+      server.listen(port, () => {
+        debug('server running on port', port);
+        server.isRunning = true;
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after((done) => {
+    if(server.isRunning) {
+      debug('server is close');
+      server.close(() => {
+        server.isRunning = false;
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  describe('testing GET /api/note/:id with valid id', function(){
+    before((done) => {
+      noteCrud.createNote({name: 'test note', content: 'test data'})
+      .then(note => {
+        debug('Created note for GET tests');
+        this.tempNote = note;
+        done();
+      })
+      .catch(() => {
+        debug('Failed to create note for GET tests.');
+        done();
+      });
+    });
+
+    it('should return a note', (done) => {
+      debug('GET note route: '+`${baseUrl}/${this.tempNote._id}`);
+      request.get(`${baseUrl}/${this.tempNote._id}`)
+      .then((res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.name).to.equal('test note');
+        done();
+      })
+      .catch(() => {
```

Comment:
Instead of using this, you can just do
```javascript
.catch(done);
```
When done is called with an error as an argument, it will fail the test. Catch will automatically call done when it's put in like this, and it will automatically pass the error in as the argument.